### PR TITLE
scan: support https URLs and stdin

### PR DESCRIPTION
**Pull request by imjasonh**
```sh
$ curl -s https://packages.wolfi.dev/os/aarch64/crossplane-provider-aws-0.39.0-r0.apk | go run ./ scan -
Will process apk from stdin
✅ No vulnerabilities found
$ go run ./ scan https://packages.wolfi.dev/os/aarch64/crossplane-provider-aws-0.39.0-r0.apk            
Will process: https://packages.wolfi.dev/os/aarch64/crossplane-provider-aws-0.39.0-r0.apk
✅ No vulnerabilities found
```

### CODE REVIEW

File: pkg/cli/scan.go [hunk 1]
```diff
@@ -127,14 +128,55 @@ func Scan() *cobra.Command {
...
+		}
+		fmt.Fprintln(os.Stderr, "Will process apk from stdin")
+		inputFilePath = t.Name()
+	} else if strings.HasPrefix(inputFilePath, "https://") {
+		// Fetch the remote apk into a temp file.
+		t, err := os.CreateTemp("", "wolfictl-scan-")
+		if err != nil {
+			return nil, fmt.Errorf("failed to create temp file for remote apk: %w", err)
+		}
+		resp, err := http.Get(inputFilePath) //nolint:gosec
```
Comment (luhring) [inline pkg/cli/scan.go:151]: Just curious what the `//nolint:gosec` is dealing with

File: pkg/cli/scan.go [hunk 2]
```diff
@@ -127,14 +128,55 @@ func Scan() *cobra.Command {
...
+		if _, err := io.Copy(t, os.Stdin); err != nil {
+			return nil, err
+		}
+		if err := t.Close(); err != nil {
+			return nil, err
+		}
+		fmt.Fprintln(os.Stderr, "Will process from stdin")
+		inputFilePath = t.Name()
+	} else if strings.HasPrefix(inputFilePath, "https://") {
+		// Fetch the remote apk into a temp file.
```
Comment (luhring) [inline pkg/cli/scan.go:146]: or SBOM, is that right?

**Review by luhring — COMMENTED** [review]
I like it! A couple minor questions for ya

File: pkg/cli/scan.go (hunk 1 above)
Comment (imjasonh) [reply pkg/cli/scan.go:151]: golangci-lint complained that a user-provided value was passed to `http.Get`, and I just don't care. 🤷

File: pkg/cli/scan.go (hunk 2 above)
Comment (imjasonh) [reply pkg/cli/scan.go:146]: Correct! Updating the comment.

File: pkg/cli/scan.go (hunk 1 above)
Comment (luhring) [reply pkg/cli/scan.go:151]: Gotcha! Cool, yeah we intend for this to be an arbitrary remote thing so that makes sense!

**Review by luhring — APPROVED** [review]
👨‍🍳 💋

**Pull request merged**